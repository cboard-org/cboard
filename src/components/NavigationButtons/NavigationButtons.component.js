import React from 'react';
import PropTypes from 'prop-types';
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
import FirstPageIcon from '@material-ui/icons/FirstPage';

import './NavigationButtons.css';

const NavigationButtons = ({
  active,
  navHistory,
  previousBoard,
  toRootBoard
}) => {
  if (!active) {
    return null;
  }

  return (
    <React.Fragment>
      {navHistory.length > 2 && (
        <div className="NavigationButton left">
          <button onClick={toRootBoard}>
            <FirstPageIcon />
          </button>
        </div>
      )}
      {navHistory.length > 1 && (
        <div className="NavigationButton right">
          <button onClick={previousBoard}>
            <ChevronLeftIcon />
          </button>
        </div>
      )}
    </React.Fragment>
  );
};

NavigationButtons.props = {
  navHistory: PropTypes.arrayOf(PropTypes.string),
  previousBoard: PropTypes.func,
  toRootBoard: PropTypes.func
};

export default NavigationButtons;
