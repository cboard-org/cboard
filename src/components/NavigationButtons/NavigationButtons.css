.NavigationButton {
  position: fixed;
  z-index: 1000;
  height: 8rem;
  width: 6rem;
  top: 40%;
  background: rgba(255, 0, 0, 0.5);
  text-align: center;
  font-size: 2rem;
  overflow: hidden;
}

.NavigationButton:hover {
  background: rgba(255, 0, 0, 0.8);
}

.NavigationButton.left {
  left: 0;
  border-top-right-radius: 50%;
  border-bottom-right-radius: 50%;
}

.NavigationButton.right {
  right: 0;
  border-top-left-radius: 50%;
  border-bottom-left-radius: 50%;
}

.NavigationButton button {
  background: transparent;
  border: 0;
  width: 100%;
  height: 100%;
  padding: 10% 0 0 0;
  color: #fff;
  cursor: pointer;
}

.NavigationButton button svg {
  width: 5rem;
  height: 5rem;
}
