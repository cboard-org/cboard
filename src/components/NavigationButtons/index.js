import NavigationButtons from './NavigationButtons.component';

export default NavigationButtons;
