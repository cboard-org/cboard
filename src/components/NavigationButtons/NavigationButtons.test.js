import React from 'react';
import { shallowMatchSnapshot } from '../../common/test_utils';
import NavigationButtons from './NavigationButtons.component';

let navHistory = ['some', 'id', 'toCheck', 'everything'];

const COMPONENT_PROPS = {
  navHistory,
  previousBoard: () => {
    navHistory.pop();
  },
  toRootBoard: () => {
    navHistory = [navHistory[0]];
  }
};

describe('NavigationButtons tests', () => {
  test('default renderer', () => {
    shallowMatchSnapshot(<NavigationButtons {...COMPONENT_PROPS} />);
  });
});
