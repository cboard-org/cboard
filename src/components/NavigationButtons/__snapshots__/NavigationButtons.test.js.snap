// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationButtons tests default renderer 1`] = `null`;
